parl/core/paddle/tests/agent_base_test.py

```diff
@@ -0,0 +1,294 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import unittest
+import paddle
+import paddle.nn as nn
+import parl
+import os
+
+
+class TestModel(parl.Model):
+    def __init__(self):
+        super(TestModel, self).__init__()
+        self.fc1 = nn.Linear(4, 256)
+        self.fc2 = nn.Linear(256, 128)
+        self.fc3 = nn.Linear(128, 1)
+
+    def predict(self, obs):
+        out = self.fc1(obs)
+        out = self.fc2(out)
+        out = self.fc3(out)
+        return out
+
+
+class ACModel(parl.Model):  # TODO: use a new file to test this model
+    def __init__(self):
+        super(ACModel, self).__init__()
+        self.actor = Actor()
+        self.critic = Critic()
+
+    def predict(self, obs):
+        return self.actor(obs)
+
+    def Q(self, obs):
+        return self.critic(obs)
+
+
+class Actor(parl.Model):
+    def __init__(self):
+        super(Actor, self).__init__()
+        self.fc1 = nn.Linear(4, 300)
+        self.fc2 = nn.Linear(300, 2)
+
+    def forward(self, x):
+        out = self.fc1(x)
+        out = self.fc2(out)
+        return out
+
+
+class Critic(parl.Model):
+    def __init__(self):
+        super(Critic, self).__init__()
+        self.fc1 = nn.Linear(4, 300)
+        self.fc2 = nn.Linear(300, 1)
+
+    def forward(self, x):
+        out = self.fc1(x)
+        out = self.fc2(out)
+        return out
+
+
+class TestAlgorithm(parl.Algorithm):
+    def __init__(self, model):
+        self.model = model
+
+    def predict(self, obs):
+        return self.model.predict(obs)
+
+    def learn(self, obs, label):
+        pred_output = self.model.policy(obs)
+        mse_loss = paddle.nn.loss.MSELoss()
+        cost = mse_loss(pred_output, label)
+        cost = paddle.reduce_mean(cost)  # TODO: paddle.mean
+        return cost
+
+
+class TestAgent(parl.Agent):
+    def __init__(self, algorithm):
+        super(TestAgent, self).__init__(algorithm)
+
+    def predict(self, obs):
+        obs = paddle.to_tensor(obs.astype(np.float32))
+        output = self.alg.predict(obs)
+        output_np = output.numpy()
+        return output_np
+
+    def learn(self, obs, label):
+        obs = paddle.to_tensor(obs.astype(np.float32))
+        label = paddle.to_tensor(label.astype(np.float32))
+        output = self.alg.learn(obs, label)
+        output_np = output.numpy()
+        return output_np
+
+
+class AgentBaseTest(unittest.TestCase):
+    def setUp(self):
+        self.model = TestModel()
+        self.alg = TestAlgorithm(self.model)
+
+    def test_agent(self):
+        agent = TestAgent(self.alg)
+        obs = np.random.random([10, 4]).astype('float32')
+        output_np = agent.predict(obs)
+        self.assertIsNotNone(output_np)
```

Comment:
We still lack of some numeric tests. Please do the computation with two numpy matrices and check the result with that given by the agent.